core/services/ardupilot_manager/mavlink_proxy/Manager.py

```diff
@@ -32,11 +36,63 @@ def available_interfaces() -> List[Type[AbstractRouter]]:
     def use(self, interface: AbstractRouter) -> None:
         self.tool = interface
 
+    def _saved_endpoints(self) -> List[Endpoint]:
+        try:
+            with open(self.endpoints_path) as f:
+                return [Endpoint(dictionary=endpoint) for endpoint in json.load(f)]
+        except IOError as e:
+            print(f"Error: Could not open endpoints cache.\n{e}")
+
+    def _save_endpoints(self, endpoints: List[Endpoint]) -> None:
+        try:
+            with open(self.endpoints_path, "w") as endpoints_file:
+                json.dump(endpoints, endpoints_file)
+        except IOError as e:
+            print(f"Error: Could not save endpoints on cache.\n{e}")
+
+    def load_endpoints(self) -> None:
+        saved_endpoints = self._saved_endpoints()
+        for endpoint in saved_endpoints:
+            if not self.tool.add_endpoint(endpoint):
+                warn(f"Endpoint {endpoint} is not valid and was not loaded.", RuntimeWarning)
```

Comment:
This entire load/save thing should be in settings

Reply:
I moved it to settings and turned it into a generic loder/saver.